test/comment.js

```diff
@@ -0,0 +1,177 @@
+        } finally {
+            assert.end();
+        }
+    });
+});
+
+
+// Exploratory test, how is whitespace treated?
+tap.test('whitespace', function (assert) {
+    assert.plan(1);
+
+    var verify = function (output) {
+        assert.equal(output.toString('utf8'), [
+                'TAP version 13',
+                '# whitespace',
+                '# ',
+                '# a',
+                '# a',
+                '# a',
+                '',
+                '1..0',
+                '# tests 0',
+                '# pass  0',
+                '',
+                '# ok',
+                ''
+        ].join('\n'));
+    };
+
+    var test = tape.createHarness();
+    test.createStream().pipe(concat(verify));
+    test('whitespace', function (t) {
+        t.comment(' ');
+        t.comment(' a');
+        t.comment('a ');
+        t.comment(' a ');
+        t.end();
+    });
+});
+
+// Exploratory test, how about passing types other than strings?
+tap.test('non-string types', function (assert) {
+    assert.plan(1);
+
+    var verify = function (output) {
+        assert.equal(output.toString('utf8'), [
+                'TAP version 13',
+                '# non-string types',
+                '# true',
+                '# false',
+                '# 42',
+                '# 6.66',
+                '# [object Object]',
+                '# [object Object]',
+                '# [object Object]',
+                '# function ConstructorFunction() {}',
+                '',
+                '1..0',
+                '# tests 0',
+                '# pass  0',
+                '',
+                '# ok',
+                ''
+        ].join('\n'));
+    };
+
+    var test = tape.createHarness();
+    test.createStream().pipe(concat(verify));
+    test('non-string types', function (t) {
+        t.comment(true);
+        t.comment(false);
+        t.comment(42);
+        t.comment(6.66);
+        t.comment({});
+        t.comment({"answer": 42});
+        function ConstructorFunction() {}
+        t.comment(new ConstructorFunction());
+        t.comment(ConstructorFunction);
+        t.end();
+    });
+});
+
+tap.test('multiline string', function (assert) {
+    assert.plan(1);
+
+    var verify = function (output) {
+        assert.equal(output.toString('utf8'), [
+                'TAP version 13',
+                '# multiline strings',
+                '# a',
+                '# b',
+                '# c',
+                '# d',
+                '',
+                '1..0',
+                '# tests 0',
+                '# pass  0',
+                '',
+                '# ok',
+                ''
+        ].join('\n'));
+    };
+
+    var test = tape.createHarness();
+    test.createStream().pipe(concat(verify));
+    test('multiline strings', function (t) {
+        t.comment([
+            'a',
+            'b',
+        ].join('\n'));
+        t.comment([
+            'c',
+            'd',
+        ].join('\r\n'));
```

Comment:
They're actually being `trim()`med in `test.js:107`: removing the inner call to `trim()` makes this part of the test fail.